File: binding_rv_adapter/src/main/java/me/lx/rv/itembindings/ItemBindingModel.java
package me.lx.rv.itembindings;

import androidx.annotation.NonNull;

import me.lx.rv.ItemBinding;


/**
 * Implement this interface on yor items to use with {@link OnItemBindModel}.
 */
public interface ItemBindingModel {
    /**
     * Set the binding variable and layout of the given view.
     * <pre>{@code
     * onItemBind.set(BR.item, R.layout.item);
     * }</pre>
     */
    void onItemBind(@NonNull ItemBinding itemBinding);
}

File: binding_rv_adapter/src/main/java/me/lx/rv/itembindings/ItemBindingModel.kt
package me.lx.rv.itembindings

import me.lx.rv.XmlItemBinding


/**
 * Implement this interface on yor items to use with [OnItemBindModel].
 */
interface ItemBindingModel {
    /**
     * Set the binding variable and layout of the given view.
     * <pre>`onItemBind.set(BR.item, R.layout.item);
    `</pre> *
     */
    fun onItemBind(itemBinding: XmlItemBinding<*>)
}
